protobuf/v1/nexus.proto

```diff
@@ -325,13 +329,18 @@ message RebuildHistoryResponse {
 }
 
 message RebuildHistoryRecord {
-  string dst_uri = 1;                       // URI of the rebuilt replica (destination)
-  string src_uri = 2;                       // URI of the source replica
-  RebuildJobState state = 3;                // Final state of the rebuild job
-  uint64 rebuilt_data_size = 4;             // Size of the data rebuilt
-  bool is_partial = 5;                      // Full or Partial rebuild
-  google.protobuf.Timestamp started_at = 6; // Start timestamp(UTC) of the rebuild job
-  google.protobuf.Timestamp ended_at = 7;   // End timestamp(UTC) of the rebuild job
+  string child_uri = 1;                      // URI of the rebuilt child (destination)
+  string src_uri = 2;                        // URI of the source replica
+  RebuildJobState state = 3;                 // final state of the rebuild job
+  uint64 blocks_total = 4;                   // total number of blocks of the replicas
+  uint64 blocks_recovered = 5;               // number of blocks recovered
+  uint64 blocks_transferred = 6;             // number of blocks for which the actual data transfer occurred
+  uint64 blocks_remaining = 7;               // number of blocks remaining to transfer
```

Comment:
Does the 'blocks_remaining' field make sense for history record?

Reply:
Or may be it is an indication for unsuccessful/aborted rebuilds where its value will be non-zero? Otherwise I'd think it'll be always zero for a successful rebuild.

Reply:
Yes it's for unsuccessful/aborted rebuilds. Normally it must be zero. Tests for rebuild should check that.